utils.go

```diff
@@ -66,3 +81,55 @@ func IsBinary(data []byte) bool {
 
 	return true
 }
+
+func loadGitattributes() (map[string]string, error) {
+	gitAttributes := map[string]string{}
+	data, err := ioutil.ReadFile(".gitattributes")
+	if err != nil {
+		if err != os.ErrNotExist {
+			log.Println(".gitattributes: " + err.Error())
+		}
+		return nil, err
+	}
+
+	if data != nil {
```

Comment:
If you actually want to avoid the code inside the if statement, you should check for `len(data) > 0`, if ioutil.ReadFile doesn't fail it could return an empty slice and an empty slice in not nil.